src/Esprima/JavascriptParser.cs

```diff
@@ -684,10 +701,14 @@ private protected virtual Expression ParsePrimaryExpression()
         switch (_lookahead.Type)
         {
             case TokenType.Identifier:
-                if ((_context.IsModule || _context.IsAsync) && "await".Equals(_lookahead.Value))
+                if ((_context.IsModule || _context.AllowAwait) && "await".Equals(_lookahead.Value))
                 {
                     TolerateUnexpectedToken(_lookahead);
                 }
+                if (_context.InClassFieldInit && "arguments".Equals(_lookahead.Value))
```

Comment:
`_context.InClassFieldInit` is never set, it always has its default value (`false`).